presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java

```diff
@@ -85,4 +99,53 @@ public static long getFileSize(Page statisticsPage, int position)
         }
         return Optional.of(manifestBuilder.build());
     }
+
+    public static Map<String, String> updatePartitionMetadataWithFileNamesAndSizes(PartitionUpdate partitionUpdate, Map<String, String> metadata)
+    {
+        ImmutableMap.Builder<String, String> partitionMetadata = ImmutableMap.builder();
+        List<FileWriteInfo> fileWriteInfos = new ArrayList<>(partitionUpdate.getFileWriteInfos());
+
+        // Sort the file infos based on fileName
+        fileWriteInfos.sort(Comparator.comparing(info -> Integer.valueOf(info.getWriteFileName())));
+
+        // Join the file names into a consolidated string
+        String fileNames = joinFileNames(fileWriteInfos.stream().map(FileWriteInfo::getWriteFileName).collect(toImmutableList()));
+
+        // Join the file sizes
+        String fileSizes = Joiner.on(COMMA).join(fileWriteInfos.stream()
+                .map(FileWriteInfo::getFileSize)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(String::valueOf)
+                .collect(toImmutableList()));
+
+        partitionMetadata.put(FILE_NAMES, fileNames);
+        partitionMetadata.put(FILE_SIZES, fileSizes);
+        partitionMetadata.putAll(metadata);
+
+        return partitionMetadata.build();
+    }
+
+    public static String joinFileNames(List<String> fileNames)
+    {
+        if (fileNames.size() == 1) {
+            return fileNames.get(0);
+        }
+
+        boolean isContinuousSequence = true;
+        int start = 0;
+        for (String name : fileNames) {
+            if (start != Integer.parseInt(name)) {
+                isContinuousSequence = false;
+                break;
+            }
+            start++;
+        }
+
+        if (isContinuousSequence) {
+            return fileNames.get(fileNames.size() - 1);
+        }
+
+        return Joiner.on(COMMA).join(fileNames);
```

Comment:
Recommend to use roaring bitmap (https://roaringbitmap.org/) for this. It will provide good compression ratio.